python/utils/bootstrap_utilities.py

```diff
@@ -391,3 +392,160 @@ def handle_error(data, proxy=None):
             proxy.close()
         except Exception:
             pass
+
+
+def logger_via_proxy(data, message):
+    """
+    Sends a log message to tk-desktop via the RPC proxy.
+
+    This function uses the RPC proxy to send log messages to the main tk-desktop process.
+    It is particularly useful for logging messages before the engine is fully initialized
+    or when direct access to the tk-desktop logging system is not available.
+
+    :param data: A dictionary containing proxy connection details, including the path
+                 to the RPC library and proxy authentication information.
+    :param message: The log message to send to tk-desktop.
+    """
+
+    sys.path.insert(0, data["core_python_path"])
+    import sgtk
+
+    # Set up logging with the rpc.
+    proxy = _create_proxy(data)
+    handler = ProxyLoggingHandler(proxy)
+    sgtk.LogManager().initialize_custom_handler(handler)
+    logger = sgtk.LogManager.get_logger(__file__)
+    logger.error(message)
+    proxy.close()
```

Comment:
Not super important, but I find this function awfully wasteful when called multiple times. 
Is there a way to keep things open for the duration of ... the process. Like with a decorator function maybe or something similar?